src/detect-smb-cmd.c

```diff
@@ -0,0 +1,117 @@
+/* Copyright (C) 2022 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+
+#include "detect.h"
+#include "detect-parse.h"
+
+#include "detect-engine.h"
+#include "detect-engine-mpm.h"
+#include "detect-engine-state.h"
+#include "detect-engine-prefilter.h"
+#include "detect-engine-content-inspection.h"
+
+#include "detect-smb-cmd.h"
+#include "rust.h"
+
+static int g_smb_cmd_list_id = 0;
+
+static void DetectSmbCmdFree(DetectEngineCtx *de_ctx, void *ptr)
+{
+    SCEnter();
+
+    SCLogDebug("smb_cmd: DetectSmbCmdFree");
+
+    if (ptr != NULL) {
```

Comment:
nit: check is already done by `rs_smb_cmd_free` itself

Reply:
Removed in https://github.com/OISF/suricata/pull/7338/commits/1d8ec738e19a8cf7597a5759ad82a1b1e4a6ea39